include/seqan/modifier/modifier_padding.h

```diff
@@ -0,0 +1,419 @@
+ * @extends ModifiedString
+ * @headerfile <seqan/modifier.h>
+ * @brief Pad characters beyond the end of a string with default value.
+ *
+ * @signature template <typename THost>
+ *            class ModifiedString<THost, ModPadding>;
+ *
+ * @tparam THost original string.
+ */
+
+template <typename THost>
+struct ModPaddingCargo
+{
+    using TSize  = typename Size<THost>::Type;
+    using TValue = typename Value<THost>::Type;
+
+    TSize   _expandedSize   = 0;
+    TSize   _remainingSteps = 0;
+    TValue  _paddedValue    = TValue();
+};
+
+// ============================================================================
+// Metafunctions
+// ============================================================================
+
+// --------------------------------------------------------------------------
+// Metafunction Reference
+// --------------------------------------------------------------------------
+
+template <typename THost>
+struct Reference<ModifiedString<THost, ModPadding> > :
+    Reference<THost>
+{};
+
+template < typename THost>
+struct Reference<ModifiedString<THost, ModPadding> const> :
+    Reference<THost const>
+{};
+
+// --------------------------------------------------------------------------
+// Metafunction Cargo                           [ModPadding ModifiedIterator]
+// --------------------------------------------------------------------------
+
+template <typename THost>
+struct Cargo<ModifiedString<THost, ModPadding> >
+{
+    using Type = ModPaddingCargo<THost>;
+};
+
+template <typename THost>
+struct Cargo<ModifiedIterator<THost, ModPadding> >
+{
+    using Type = ModPaddingCargo<typename Container<THost>::Type>;
+};
+
+// --------------------------------------------------------------------------
+// Metafunction Iterator                          [ModPadding ModifiedString]
+// --------------------------------------------------------------------------
+
+template <typename THost>
+struct Iterator<ModifiedString<THost, ModPadding>, Standard>
+{
+    typedef ModifiedIterator<typename Iterator<THost, Rooted>::Type, ModPadding> Type;
+};
+
+template <typename THost>
+struct Iterator<ModifiedString<THost, ModPadding> const, Standard>
+{
+    typedef ModifiedIterator<typename Iterator<THost, Rooted>::Type, ModPadding> Type;
+};
+
+// --------------------------------------------------------------------------
+// Metafunction DefaultIteratorSpec               [ModReverse ModifiedString]
+// --------------------------------------------------------------------------
+
+template <typename THost>
+struct DefaultIteratorSpec< ModifiedString<THost, ModPadding> >
+{
+    typedef Rooted Type;
+};
+
+// --------------------------------------------------------------------------
+// Metafunction AllowsFastRandomAccess
+// --------------------------------------------------------------------------
+
+template <typename THost>
+struct AllowsFastRandomAccess<ModifiedString<THost, ModPadding> > :
+    AllowsFastRandomAccess<THost>
+{};
+
```

Comment:
Apparently it is not related to the ```RandomAccessIteratorConcept``` at all but used solely for index construction. But I see the point, that this might not be relevant for padded strings.